patches/glib/0001-Small-fixes.patch

```diff
@@ -0,0 +1,160 @@
+From e3257b91e4920fbf4753452aa05d5b9941024c8e Mon Sep 17 00:00:00 2001
+From: no92 <no92.mail@gmail.com>
+Date: Sun, 26 Apr 2020 01:05:06 +0200
+Subject: [PATCH] Small fixes
+
+---
+ gio/gnetworking.h.in       |  2 ++
+ gio/gsocket.c              | 33 +++++++++++++++++++++++++++++++++
+ gio/gthreadedresolver.c    |  7 +++++--
+ gio/meson.build            |  4 ++--
+ gio/xdgmime/xdgmimecache.c |  2 +-
+ glib/glib-init.c           |  1 +
+ glib/gstrfuncs.c           |  1 +
+ 7 files changed, 45 insertions(+), 5 deletions(-)
+
+diff --git a/gio/gnetworking.h.in b/gio/gnetworking.h.in
+index f9582b9..c3a5024 100644
+--- a/gio/gnetworking.h.in
++++ b/gio/gnetworking.h.in
+@@ -47,7 +47,9 @@
+ #include <net/if.h>
+ 
+ #include <arpa/inet.h>
++#if __has_include(<arpa/nameser.h>)
+ #include <arpa/nameser.h>
++#endif
+ @NAMESER_COMPAT_INCLUDE@
+ 
+ #ifndef T_SRV
+diff --git a/gio/gsocket.c b/gio/gsocket.c
+index 66073af..9c2dbd8 100644
+--- a/gio/gsocket.c
++++ b/gio/gsocket.c
+@@ -81,6 +81,39 @@
+ #include "gwin32networking.h"
+ #endif
+ 
++#if defined(__managarm__)
++#ifndef IP_TTL
++#define IP_TTL 0
++#endif
++#ifndef IP_MULTICAST_LOOP
++#define IP_MULTICAST_LOOP 0
++#endif
++#ifndef IP_MULTICAST_TTL
++#define IP_MULTICAST_TTL 0
++#endif
++#if !defined(IP_ADD_MEMBERSHIP) && !defined(IP_DROP_MEMBERSHIP) && !defined(HAVE_IP_MREQN)
++struct ip_mreq
++{
++	struct in_addr imr_multiaddr;
++	struct in_addr imr_interface;
++};
++#endif
++#ifndef IP_ADD_MEMBERSHIP
++#define IP_ADD_MEMBERSHIP 0
++#endif
++#ifndef IP_DROP_MEMBERSHIP
++#define IP_DROP_MEMBERSHIP 0
++#endif
++#ifndef SO_NREAD
++#define SO_NREAD 0
++#endif
++#ifndef CMSG_DATA
++#define CMSG_DATA(x) ((unsigned char*) x)
++#define CMSG_FIRSTHDR(x) ((struct cmsghdr *) NULL)
++#define CMSG_NXTHDR(x, y) ((struct cmsghdr *) NULL)
++#endif
++#endif
```

Comment:
These can probably be (mostly?) fixed in mlibc. Maybe that's something @Dennisbonke can integrate into his mlibc PR?